types/wlr_buffer.c

```diff
@@ -194,8 +168,13 @@ static void client_buffer_resource_handle_destroy(struct wl_listener *listener,
 	// which case we'll read garbage. We decide to accept this risk.
 }
 
-struct wlr_client_buffer *wlr_client_buffer_import(
-		struct wlr_renderer *renderer, struct wl_resource *resource) {
+struct wlr_buffer *wlr_buffer_import(struct wlr_renderer *renderer,
+		struct wl_resource *resource) {
+	if (!resource) {
```

Comment:
I think we can just let `wlr_resource_is_buffer` segfault instead. It's a caller error to provide a NULL `wl_resource`.

Segfaults are sometimes better than gracefully handling errors, because they complain loudly and result in user bug reports. Error logs can go unnoticed.

Reply:
I merged it with the other assert to show that we know this is supposed to segfault.